### PR TITLE
[layer] Update GRU for layer_v2

### PR DESCRIPTION
**Pull request by kparichay**
Update GRU implementation using layer_v2 design.
Update and enable corresponding unittests.

Signed-off-by: Parichay Kapoor <kparichay@gmail.com>

### CODE REVIEW

File: nntrainer/layers/gru.h [hunk 1]
```diff
@@ -59,54 +58,52 @@ class GRULayer : public LayerV1 {
...
   /**
    * @copydoc Layer::getType()
    */
   const std::string getType() const override { return GRULayer::type; };
 
-  using LayerV1::setProperty;
+  /**
+   * @copydoc Layer::supportBackwarding()
+   */
+  bool supportBackwarding() const { return true; }
```
Comment (zhoonit) [inline nntrainer/layers/gru.h:93]: override here?

**Review by zhoonit — APPROVED** [review]
LGTM :)

File: nntrainer/layers/gru.h (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/gru.h:93]: i will add override when rebasing this.

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/layers/gru.h (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/gru.h]: done.

**Pull request merged**